### PR TITLE
UIIN-1226 Confirm items before moving to different holdings and/or instances

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIIN-1226

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/Instance/MoveHoldingContext/MoveHoldingContext.js [hunk 1]
```diff
@@ -150,30 +150,41 @@ const MoveHoldingContext = ({
...
-    } else {
-      setIsMoving(true);
-      moveItems(to, items)
-        .finally(() => {
-          setIsMoving(false);
-        });
+    toggleMoveModal(true);
+  }, [selectedHoldingsMap, selectedItemsMap]);
+
+  const getMovingMessage = useCallback(() => {
```
Comment (NikitaSedyx) [inline src/Instance/MoveHoldingContext/MoveHoldingContext.js:164]: memo?

File: src/Instance/MoveHoldingContext/MoveHoldingContext.js [hunk 2]
```diff
@@ -33,18 +37,30 @@ const MoveHoldingContext = ({
...
     toggleMoveModal(false);
     setIsMoving(true);
 
-    moveHoldings(dragToId, movingItems)
-      .finally(() => {
-        setIsMoving(false);
-      });
+    if (isHoldingMoved) {
+      moveHoldings(dragToId, movingItems)
+        .finally(() => {
```
Comment (NikitaSedyx) [inline src/Instance/MoveHoldingContext/MoveHoldingContext.js:50]: copy paste

File: src/Instance/MoveHoldingContext/MoveHoldingContext.js [hunk 3]
```diff
@@ -215,15 +226,7 @@ const MoveHoldingContext = ({
...
-          message={
-            intl.formatMessage(
-              { id: 'ui-inventory.moveItems.modal.message' },
-              {
-                count: movingItems.length,
-                targetName: <b>{rightInstance.id === dragToId ? rightInstance.title : leftInstance.title}</b>
-              }
-            )
-          }
+          message={getMovingMessage()}
```
Comment (aliaksei-chumakou) [inline src/Instance/MoveHoldingContext/MoveHoldingContext.js:229]: this usage semantically works with `useMemo` since you actually don't need a callback

**Review by aliaksei-chumakou — APPROVED** [review]

File: src/Instance/MoveHoldingContext/MoveHoldingContext.js [hunk 4]
```diff
@@ -150,30 +147,41 @@ const MoveHoldingContext = ({
...
     if (!items.length) {
       items.push(from);
     }
     setMovingItems(items);
-    if (isHolding) {
-      toggleMoveModal(true);
+    toggleMoveModal(true);
+  }, [selectedHoldingsMap, selectedItemsMap]);
+
+  const getMovingMessage = useMemo(() => {
```
Comment (NikitaSedyx) [inline src/Instance/MoveHoldingContext/MoveHoldingContext.js:161]: it's not a function

**Review by ViktorSoroka07 — APPROVED** [review]

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**